obsah/__init__.py

```diff
@@ -36,6 +36,50 @@
 Variable = namedtuple('Variable', ['name', 'parameter', 'help_text', 'action', 'type', 'choices'])
 
 
+class AppendUniqueAction(argparse.Action):
+    """
+    Custom argparse action that appends values but ensures uniqueness.
+    Similar to action='append', but duplicate values are ignored.
+    """
+    def __init__(self, option_strings, dest, default=None, **kwargs):
+        if default is None:
+            default = []
+        super().__init__(option_strings, dest, default=default, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        items = getattr(namespace, self.dest, None)
+        if items is None:
+            items = []
+        # Create a new list to avoid modifying the default
+        items = list(items)
+        # Only append if the value is not already in the list
+        if values not in items:
```

Comment:
This doesn't use `set`. Is that to maintain order (which IIRC `set` doesn't guarantee)?

Reply:
More to maintain API compatiblity with `append` which uses a list.

I could do `list(set(items))`, but that felt silly too.